runner/src/stream-handler/stream-handler.ts

```diff
@@ -67,8 +67,8 @@ export default class StreamHandler {
     indexer.setStoppedStatus().catch((e) => {
       this.logger.error('Failed to set stopped status for indexer', e);
     });
-
-    const streamErrorLogEntry = LogEntry.systemError(`Encountered error processing stream: ${this.indexerConfig.redisStreamKey}, terminating thread\n${error.toString()}`, this.executorContext.block_height);
+    const errorAsString = error.toString() === '[object Object]' ? JSON.stringify(error) : error.toString();
```

Comment:
```suggestion
    const errorMessage = error instanceof Error ? error.toString() : JSON.stringify(error);
```

I believe `[Object object]` would only appear if we are throwing an Object somewhere. I think we can just change this to the above to ensure we are only calling `toString()` on `Error`s.

Reply:
Tested this and it seems to work. I think this is cleaner so I'll go with this. Thanks!